File: backend/src/main/java/com/tododuk/global/security/SecurityUser.java
package com.tododuk.global.security;

import lombok.Getter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;

import java.util.Collection;

public class SecurityUser extends User {
    @Getter
    private int id;
    @Getter
    private String email;
    //username 대신 email 사용
    public SecurityUser(
            int id,
            String email,
            String password,
            Collection<? extends GrantedAuthority> authorities
    ) {
        super(email, password, authorities);
        this.id = id;
        this.email = email;
    }

}

File: backend/src/main/java/com/tododuk/global/security/SecurityUser.kt
package com.tododuk.global.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.User
import org.springframework.security.oauth2.core.user.OAuth2User

class SecurityUser(
    val id: Int,
    val email: String,
    password: String,
    authorities: Collection<out GrantedAuthority>
) : User(email, password, authorities), OAuth2User {
    override fun getAttributes(): Map<String, Any> = emptyMap()
    override fun getName(): String = email
}
